builder_test.go

```diff
@@ -209,6 +210,38 @@ RUN echo $FOO $BAR`))
 	}
 }
 
+// Test if `FROM some-${SOME-BUILT-IN-ARG}` args gets resolved correctly.
+func TestArgResolutionOfDefaultVariables(t *testing.T) {
+	// Get architecture from host
+	var localspec = platforms.DefaultSpec()
+	for _, tc := range []struct {
+		name         string
+		args         map[string]string
+		expectedFrom string
+	}{
+		{name: "override", args: map[string]string{"FOO": "bar"}, expectedFrom: "platform-"+localspec.Architecture},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			n, err := ParseDockerfile(strings.NewReader(`ARG FOO=latest
+ARG BAR=baz
+FROM platform-${TARGETARCH}
+ARG BAZ=banana
+RUN echo $FOO $BAR`))
+			if err != nil {
+				t.Fatal(err)
+			}
+			b := NewBuilder(tc.args)
+			from, err := b.From(n)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if from != tc.expectedFrom {
+				t.Fatalf("expected %s, got %s", tc.expectedFrom, from)
+			}
+		})
+	}
+}
+
```

Comment:
Do we have a test that also covers this?

`also allows FROM ${SOME_DEFAULT_OR_USER_ARG} in general.`

Reply:
@TomSweeneyRedHat I think test in this PR should cover that as well since it performs test for `$TARGETARCH` which is one of the DEFAULT ARGS but i can add another test for user args as well.

Reply:
@TomSweeneyRedHat Added additional test case.